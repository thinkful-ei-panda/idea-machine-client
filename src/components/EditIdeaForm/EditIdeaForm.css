fieldset{
  border:none;
}

fieldset .formInputContainer{
  display:flex;
  flex-direction: column;
}

label{
  text-align: left;
  padding-top: 18px;
  padding-bottom: 8px;
}

.buttonContainer{
  padding-top: 15px;
}