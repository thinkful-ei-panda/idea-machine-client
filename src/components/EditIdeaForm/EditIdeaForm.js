import React from 'react'
import config from '../../config'
import TokenService from '../../services/token-service'

class EditIdeaForm extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
    editIdeaValues:{}
  }

  state = {
    error:null
  }

  handleEditIdeaFormSubmit = (ev,id) => {
    ev.preventDefault()
    this.setState({error:null})
    const {title,content} = ev.target

    fetch(`${config.API_ENDPOINT}/ideas/${id}`, {
      method: 'PATCH',
      headers: {
        'content-type':'application/json',
        'Authorization':`bearer ${TokenService.getAuthToken()}`,
      },
      body:JSON.stringify({
        title:title.value,
        content:content.value
      })
    })
    .then(res => (!res.ok)
    ?res.json().then(e => Promise.reject(e))
    :res)
    .then(() => {
      this.props.history.push('/my-ideas')
    })
    .catch(error => {
      this.setState({error:error.error})
    })    
  }

  render(){
    const {title, content,id} = this.props.editIdeaValues

    return (
      <div>
        <form onSubmit={ev =>this.handleEditIdeaFormSubmit(ev,id)}>
          <legend></legend>
          <fieldset>
            <div>
              <label htmlFor='title'>Title</label>
              <input id='title' name='title' defaultValue={title}/>
            </div>            
            <div>
              <label htmlFor='content'>Content</label>
              <textarea id='content' name='content' defaultValue={content}/>
            </div>           
            <button>Finish Edit</button>
          </fieldset>
        </form>
        {this.state.error && <div>{this.state.error}</div>}
      </div>
    )
  }
}

export default EditIdeaForm