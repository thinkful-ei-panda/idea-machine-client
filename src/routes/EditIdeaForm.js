import React from 'react'
import EditIdeaForm from '../components/EditIdeaForm/EditIdeaForm'

class EditIdeaPage extends React.Component {

  render() {
    return (
      <div className='AddIdeaPage'>
        <h2>Add Idea</h2>
        <EditIdeaForm {...this.props}/>
      </div>
    )
  }
}

export default EditIdeaPage