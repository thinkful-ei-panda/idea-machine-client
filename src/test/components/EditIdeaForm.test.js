import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import EditIdeaForm from '../../components/EditIdeaForm/EditIdeaForm';
describe(`EditIdeaForm Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <EditIdeaForm />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <EditIdeaForm />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})