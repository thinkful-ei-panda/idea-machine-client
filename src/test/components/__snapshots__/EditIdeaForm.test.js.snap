// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditIdeaForm Component Snapshot test Renders the UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <legend>
      Edit Idea Form
    </legend>
    <fieldset>
      <div
        className="column"
      >
        <div
          className="formInputContainer"
        >
          <label
            htmlFor="title"
          >
            Title
          </label>
          <input
            id="title"
            name="title"
          />
        </div>
        <div
          className="formInputContainer"
        >
          <label
            htmlFor="content"
          >
            Content
          </label>
          <textarea
            id="content"
            name="content"
          />
        </div>
        <div
          className="buttonContainer"
        >
          <button>
            Finish Edit
          </button>
        </div>
      </div>
    </fieldset>
  </form>
</div>
`;
